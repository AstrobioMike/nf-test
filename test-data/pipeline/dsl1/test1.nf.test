nextflow_pipeline {

  name "Test test1.nf"

  script "test-data/pipeline/dsl1/test1.nf"

  test("Should create 5 output files") {

   when {
     params.lukas = 27
     params.forer = "22"
     params.outdir = "output"
   }

   then {
     assert workflow.success
     assert params.lukas == 27
     for (def i in 1..5){
       def file = new File("${params.outdir}/${i}_27.txt");
       assert file.exists()
       assert file.text == "lukas forer\n"
     }
     assert snapshot([
     	"${params.outdir}/1_27.txt",
     	"${params.outdir}/2_27.txt",
     	"${params.outdir}/3_27.txt",
     	"${params.outdir}/4_27.txt",
     	"${params.outdir}/5_27.txt",
     ]
     ).match("process_test_1")
   }

  }


  test("Should fail if outdir is missing") {

    //debug true

    when {
      params.lukas = 27
      params.forer = "22"
    }

    then {
      assert workflow.failed
      assert params.lukas == 27
      assert params.forer == "22"
    }

  }

}
