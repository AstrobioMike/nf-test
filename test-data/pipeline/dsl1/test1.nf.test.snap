{
    "process_test_1": {
        "content": [
            "1_27.txt:086e29e2872195015ce2d3104909b20f",
            "2_27.txt:086e29e2872195015ce2d3104909b20f",
            "3_27.txt:086e29e2872195015ce2d3104909b20f",
            "4_27.txt:086e29e2872195015ce2d3104909b20f",
            "5_27.txt:086e29e2872195015ce2d3104909b20f"
        ]
    }
}